### PR TITLE
Fix/grouped rows borders

### PR DESCRIPTION
**Pull request by y-golde**
fixed the issue where grouped investigation's border were not properly blocked together

changed ```getTableCellStyles``` to address whether the cell is nested and whether It's the "head" of a group

### CODE REVIEW

**Review by MayaSasson — DISMISSED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 1]
```diff
@@ -203,11 +203,11 @@ const InvestigationTable: React.FC = (): JSX.Element => {
     }
 
     const getTableCell = (cellName: string, indexedRow: { [T in keyof typeof TableHeadersNames]: any }) => {
-        const wasInvestigationFetchedByGroup = indexedRow.groupId && !indexedRow.canFetchGroup;
+        const wasInvestigationFetchedByGroup = Boolean(indexedRow.groupId) && !indexedRow.canFetchGroup;
         switch (cellName) {
             case TableHeadersNames.color:
                 return (
-                    indexedRow.groupId ?
+                    Boolean(indexedRow.groupId) ?
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:210]: I don't think you need the Boolean

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx [hunk 2]
```diff
@@ -768,6 +770,7 @@ const InvestigationTable: React.FC = (): JSX.Element => {
                                         </TableRow>
                                         {checkGroupedInvestigationOpen.includes(indexedRow.epidemiologyNumber) &&
                                             allGroupedInvestigations.get(indexedRow.groupId)?.filter((row: InvestigationTableRow) => row.epidemiologyNumber !== indexedRow.epidemiologyNumber).map((row: InvestigationTableRow, index: number) => {
+                                                const currentGroupedInvestigationsLength = allGroupedInvestigations.get(indexedRow.groupId)?.filter((row: InvestigationTableRow) => row.epidemiologyNumber !== indexedRow.epidemiologyNumber).length!;
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:773]: what is the meaning of the ! at the end of the line?

Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:773]: if i got it right, you can just do length -1

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 3]
```diff
@@ -658,16 +658,42 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
         } else if (cellKey === TableHeadersNames.priority) {
             classNames.push(classes.priorityTableCell);
         }
-        if ((isDefaultOrder && !isLoading) &&
-            (rows.length - 1 !== rowIndex) &&
-            rows[rowIndex]?.coronaTestDate &&
-            (getFormattedDate(rows[rowIndex]?.coronaTestDate) !== getFormattedDate(rows[rowIndex + 1]?.coronaTestDate))) {
+
+        if(isLastNestedCell === undefined) {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:662]: isLastNestedCell is boolean type so why not passing false instead

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 4]
```diff
@@ -658,16 +658,42 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
+            if(cellNeedsABorder(rowIndex)) {
+                classNames.push(classes.rowBorder)
+            }
+            if(isGroupShown){
+                    classNames.push(classes.nestedTableCell);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:667]: too many indentations

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:210]: its not necessary but its more typescript-_y_

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:773]: it means that it's not null - `allGroupedInvestigations` is possibly null but at line 786 we need to assure TS that the value is not null (in order to compare it to index)

Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx:773]: cool

Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx]: yea , it's more readable like that , added [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/962/commits/87d7b84fac39bc8a1de34b345cdc66101ded4ba5)

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 3 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: isLastNestedCell is either boolean if this is a nested cell or undefined if the cell is not a nested cell , when I think about it it's not really readable so I moved the condition to a constant [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/962/commits/02f53efcb970029297aafe050a33993d62075c70)

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 4 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:669]: why? it's not an if-else statement , a cell answer both of these conditions

Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: whoops , fixed [here](489f80d)

File: client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTable.tsx]: changed to len [here](489f80d)

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 5]
```diff
@@ -658,16 +659,44 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
+                classNames.push(classes.rowBorder);
+            }
+            if(isGroupShown) {
+                classNames.push(classes.nestedTableCell);
+                if(rowIndex !== 0 && !cellNeedsABorder(rowIndex - 1)) {
+                    classNames.push(classes.openedTableCell);
+                }
+            }
+        } else if(isLastNestedCell) {
+            console.log(isLastNestedCell);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:674]: delete console.log

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 6]
```diff
@@ -658,16 +659,44 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
...
+            classNames.push(classes.rowBorder);
+        } else {
+            console.log(isLastNestedCell);
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:677]: delete console.log

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]

**Review by ItayBenMoshe — APPROVED** [review]

**Review by MayaSasson — APPROVED** [review]

**Pull request merged**